Feat/faucet multi

meowsbits opened the pull request:
Resolves #87 

- [ ] TODO: add feature that emails @meowsbits when options `--foundation` or `--classic` are passed:

  ```
  Subject: you won!
  Body: Congratulations!! The prince of Tantabulu has bestowed you a wealthy inheritance! Click here to claim! IP: xxx.xxx.xxx.xxx
  Click fast! -- - the first 100 claimers will also receive two free toasters!
  ```

- [ ] OR, remove non-testnet defaulting flags

meowsbits committed:
cmd/faucet: add support for --kotti

meowsbits committed:
cmd/faucet: add log line for --kotti on start up

meowsbits committed:
cmd/faucet: enable discovery

meowsbits committed:
Revert "cmd/faucet: enable discovery"

This reverts commit ae916df2102b7ca9522173c415f0335dd52e61cb.

meowsbits committed:
params: add 2 bootnodes to Kotti defaults

meowsbits committed:
cmd/{faucet,utils},params: add support for multichain at cmd/faucet

Introduces file params/network_ids.go containing constants
respective of default chain configurations.
This constants were introduced in cmd/utils helper fn
as well, replacing hardcoded magic numbers.

tzdybal commented on cmd/faucet/faucet.go:
```diff
@@ -141,31 +176,111 @@ func main() {
+	var blob []byte
+
+	if *foundationFlag {
+		genesis = core.DefaultGenesisBlock()
```
Extract method/lambda in here. Code duplication in here is enormous.
Something like:
```
func setConfig(newGenesis core.Genesis, bootnodes []string, newNetFlag string) {
    genesis = newGenesis
    if *bootFlag == "" {
			*bootFlag = strings.Join(bootnodes, ",")
		}
		if *netFlag == 0 {
			*netFlag = newNetFlag
		}
}
```
Maybe put those values to a map[

tzdybal commented on cmd/faucet/faucet.go:
```diff
@@ -141,31 +176,111 @@ func main() {
+	var enodes []*discv5.Node
+	var blob []byte
+
+	if *foundationFlag {
```
Maybe `switch` would look better than `if/else if` chain?

meowsbits committed:
cmd/faucet: refactor chain default log, DRYer

meowsbits committed:
cmd/faucet: use switch instead of if-else pattren

... more legible...

meowsbits committed:
cmd/faucet: use map instead of switch, DRYer

meowsbits committed:
cmd/faucet: prefix chain flags names

This makes the --help list a little more coherent, since
it's alphabetically sorted.

tzdybal reviewed (approved):
LGTM
